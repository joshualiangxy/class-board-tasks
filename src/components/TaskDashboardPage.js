import React from 'react';
import { connect } from 'react-redux';

export const TaskDashboardPage = () => (
  <div>
    <h1>This is the task dashboard page</h1>
  </div>
);

const mapDispatchToProps = dispatch => ({});

export default connect(undefined, mapDispatchToProps)(TaskDashboardPage);
