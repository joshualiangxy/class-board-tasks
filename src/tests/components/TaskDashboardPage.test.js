import React from 'react';
import { shallow } from 'enzyme';
import TaskDashboardPage from '../../components/TaskDashboardPage';

const wrapper = shallow(<TaskDashboardPage />);

describe('render', () => {
  it('should render TaskDashboardPage', () =>
    expect(wrapper).toMatchSnapshot());
});

describe('add task button', () => {
  it('should open AddTaskModal on click', () => {
    expect(wrapper.find('Connect(AddTaskModal)').prop('isOpen')).toBe(false);

    wrapper.find('button').prop('onClick')();

    expect(wrapper.find('Connect(AddTaskModal)').prop('isOpen')).toBe(true);
  });
});
