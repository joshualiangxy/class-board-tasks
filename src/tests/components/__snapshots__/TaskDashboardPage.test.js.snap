// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render TaskDashboardPage 1`] = `
<div>
  <h1>
    Dashboard
  </h1>
  <Connect(TaskListFilters) />
  <Connect(TaskList) />
  <button
    onClick={[Function]}
  >
    Add Task
  </button>
  <Connect(AddTaskModal)
    isOpen={false}
    onRequestClose={[Function]}
  />
</div>
`;
